node/pkg/common/chainlock.go

```diff
@@ -272,6 +429,105 @@ func UnmarshalMessagePublication(data []byte) (*MessagePublication, error) {
 	return msg, nil
 }
 
+// UnmarshalBinary implements the BinaryUnmarshaler interface for MessagePublication.
+func (msg *MessagePublication) UnmarshalBinary(data []byte) error {
+	// Calculate minimum required length for the fixed portion
+	// (excluding variable-length fields: TxID and Payload)
+
+	// Initial check for minimum data length
+	if len(data) < marshaledMsgSizeMin {
+		return ErrInputSize{Msg: "data too short"}
+	}
+
+	// Set up deserialization
+	be := binary.BigEndian
+	pos := 0
+
+	// TxID length
+	txIDLen := uint8(data[pos])
+	pos++
+
+	// Check if we have enough data for TxID and the rest
+	if len(data) < marshaledMsgSizeMin+int(txIDLen) {
+		return ErrInputSize{Msg: "data too short"}
+	}
+
+	// Read TxID
+	msg.TxID = make([]byte, txIDLen)
+	copy(msg.TxID, data[pos:pos+int(txIDLen)])
+	pos += int(txIDLen)
+
+	// Timestamp
+	timestamp := be.Uint64(data[pos : pos+8])
+	// Nanoseconds are not serialized
+	//nolint:gosec // uint64 and int64 have the same number of bytes, and Unix time won't be negative.
+	msg.Timestamp = time.Unix(int64(timestamp), 0)
+	pos += 8
+
+	// Nonce
+	msg.Nonce = be.Uint32(data[pos : pos+4])
+	pos += 4
+
+	// Sequence
+	msg.Sequence = be.Uint64(data[pos : pos+8])
+	pos += 8
+
+	// ConsistencyLevel
+	// TODO: This could be validated against the valid range of values for ConsistencyLevel.
+	msg.ConsistencyLevel = data[pos]
+	pos++
+
+	// EmitterChain
+	msg.EmitterChain = vaa.ChainID(be.Uint16(data[pos : pos+2]))
+	pos += 2
+
+	// EmitterAddress
+	copy(msg.EmitterAddress[:], data[pos:pos+32])
+	pos += 32
+
+	// IsReobservation
+	if !validBinaryBool(data[pos]) {
+		return ErrInvalidBinaryBool
+	}
+	msg.IsReobservation = data[pos] != 0
+	pos++
+
+	// Unreliable
+	if !validBinaryBool(data[pos]) {
+		return ErrInvalidBinaryBool
+	}
+	msg.Unreliable = data[pos] != 0
+	pos++
+
+	// verificationState
+	if data[pos] > NumVariantsVerificationState {
```

Comment:
Should this be `>=` as the highest value in the enum is 5?

Reply:
This should be fixed now and I added a unit test